File: solutions/c/triangle/1/triangle.c
#include "triangle.h"

bool is_equilateral(triangle_t sides) {
    double a = sides.a, b = sides.b, c = sides.c;
    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
        return false;
    }
    return (a == b && b == c);
}

bool is_isosceles(triangle_t sides) {
    double a = sides.a, b = sides.b, c = sides.c;
    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
        return false;
    }
    return (a == b || a == c || b == c);
}

bool is_scalene(triangle_t sides) {
    double a = sides.a, b = sides.b, c = sides.c;
    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
        return false;
    }
    return (a != b && a != c && b != c);
}

File: solutions/javascript/triangle/1/triangle.js
//
// This is only a SKELETON file for the 'Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Triangle {
  constructor(...sides) {
    let [a, b, c] = sides;
    this.a = a;
    this.b = b;
    this.c = c;
    this.isValid = (a > 0 && b > 0 && c > 0) && 
                   (a + b >= c) && (a + c >= b) && (b + c >= a) &&
                   (a + b + c > 0); 
    this.isRealTriangle = this.isValid && (a + b > c && a + c > b && b + c > a);
  }

  get isEquilateral() {
    if (!this.isRealTriangle) return false;
    return this.a === this.b && this.b === this.c;
  }

  get isIsosceles() {
    if (!this.isRealTriangle) return false;
    return this.a === this.b || this.a === this.c || this.b === this.c;
  }

  get isScalene() {
    if (!this.isRealTriangle) return false;
    return this.a !== this.b && this.a !== this.c && this.b !== this.c;
  }
}
